### PR TITLE
[chore] [internal/aws] Change TestMapWithExpiry to explicitly test cleanup

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** Changes the behavior of the specific test case to be more aligned with its behavior pre #25066

**Link to tracking Issue:** #25095

### CODE REVIEW

**Comment by bryan-aguilar** [comment]
@dmitryax

File: internal/aws/metrics/metric_calculator_test.go [hunk 1]
```diff
@@ -130,14 +139,14 @@ func TestMapWithExpiryCleanup(t *testing.T) {
 	assert.Equal(t, 1, store.Size())
 	store.Unlock()
 
-	time.Sleep(time.Second + time.Millisecond)
+	time.Sleep(time.Second * 2)
```
Comment (dmitryax) [inline internal/aws/metrics/metric_calculator_test.go:142]: This is a significant increase. Why is this needed?

Comment (bryan-aguilar) [reply internal/aws/metrics/metric_calculator_test.go]: There's actually no need. I reduced it even further by dropping the TTL and the sleep time together.

**Review by dmitryax — APPROVED** [review]

File: internal/aws/metrics/metric_calculator_test.go [hunk 2]
```diff
@@ -130,14 +139,14 @@ func TestMapWithExpiryCleanup(t *testing.T) {
 	assert.Equal(t, 1, store.Size())
 	store.Unlock()
 
-	time.Sleep(time.Second + time.Millisecond)
+	time.Sleep(time.Millisecond * 2)
```
Comment (Aneurysm9) [inline internal/aws/metrics/metric_calculator_test.go:142]: Is this sleep still needed?

**Review by Aneurysm9 — APPROVED** [review]

File: internal/aws/metrics/metric_calculator_test.go (hunk 2 above)
Comment (bryan-aguilar) [reply internal/aws/metrics/metric_calculator_test.go:142]: Is there ever a chance that `time.Now()` can return the same value? If so that would put a requirement for the sleep.
